lib/node_modules/@stdlib/assert/is-domain-name/test/test.js

```diff
@@ -0,0 +1,121 @@
+/*
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var isDomainName = require( './../lib' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.equal( typeof isDomainName, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function returns `true` if provided a valid domain name', function test( t ) {
+	var bool;
+
+	bool = isDomainName( 'example.com' );
+	t.equal( bool, true, 'returns true' );
+
+	bool = isDomainName( 'foo.example.com' );
+	t.equal( bool, true, 'returns true' );
+
+	bool = isDomainName( 'foo.bar.example.com' );
+	t.equal( bool, true, 'returns true' );
+
+	bool = isDomainName( 'foo.bar.baz.example.com' );
+	t.equal( bool, true, 'returns true' );
+
+	bool = isDomainName( 'foo.bar.baz.qux.example.com' );
+	t.equal( bool, true, 'returns true' );
+
+	bool = isDomainName( 'foo.bar.baz.qux.quux.example.com' );
+	t.equal( bool, true, 'returns true' );
+
+	t.end();
+});
+
+tape( 'the function returns `false` if provided an invalid domain name', function test( t ) {
+	var values;
+	var bool;
+	var i;
+
+	values = [
```

Comment:
@Planeshifter To exercise the `2` and `255` character limits, should add tests which exceed these bounds. One for an empty string, another for a single char string, and another for a string longer than 255 chars.